estimation/observations.R

```diff
@@ -3,213 +3,88 @@ library(strafica)
-df$licence = NA
-df$licence = ifelse(people$T_AJOKORTTI == 1, 1, 0)
-df$licence = ifelse(df$minor, 0, df$licence)
-df$licence = ifelse(is.na(people$T_AJOKORTTI), 9, df$licence)
-df$car_user = NA
-df$car_user = ifelse(df$cars_owned %in% c(1,2) &
-                         df$licence == 1 &
-                         people$T_HAKULJ == 1 &
-                         !df$minor, 1, 0)
-df$car_user = ifelse(is.na(df$car_user), 9, df$car_user)
-df$employed = NA
-df$employed = ifelse(people$T_ANSIOTYO %in% c(1,2), 1, 0)
-df$employed = ifelse(df$minor & is.na(people$T_ANSIOTYO), 0, df$employed)
-df$employed = ifelse(is.na(people$T_ANSIOTYO), 9, df$employed)
-df = unpick(df, minor)
-df$children = NA
-df$children = ifelse(people$T_0_6V > 0, 1, 0)
-df$children = ifelse(is.na(people$T_0_6V), 9, df$children)
-df$female = ifelse(people$T_SUKUPUOLI == 2, 1, 0)
-df$age_7_17 = ifelse(people$T_IKA >= 7 & people$T_IKA <= 17, 1, 0)
-df$age_18_29 = ifelse(people$T_IKA >= 18 & people$T_IKA <= 29, 1, 0)
-df$age_30_49 = ifelse(people$T_IKA >= 30 & people$T_IKA <= 49, 1, 0)
-df$age_50_64 = ifelse(people$T_IKA >= 50 & people$T_IKA <= 64, 1, 0)
-df$age_65 = ifelse(people$T_IKA >= 65, 1, 0)
-df$age_missing = ifelse(is.na(people$T_IKA) | people$T_IKA < 7, 9, 0)
-
-m = match(people$rsij2019, zones$zone_orig)
-df$rzone = zones$zone[m]
-df$rzone_capital_region = ifelse(zones$capital_region[m], 1, 0)
-df$rzone_surrounding_municipality = ifelse(zones$surrounding_municipality[m], 1, 0)
-df$rzone_peripheral_municipality = ifelse(zones$peripheral_municipality[m], 1, 0)
-
-background_information = rbind_list(background_information, df)
-
-# Add personal data to tours
-
-stopifnot(length(unique(background_information$pid)) == nrow(background_information))
-observations = leftjoin(observations, background_information)
-
-# Divide data into two models
-
-m = match(observations$rzone, zones$zone)
-hs15_model = (zones$capital_region[m] | zones$surrounding_municipality[m]) &
-    observations$izone %in% zones$zone &
-    observations$jzone %in% zones$zone &
-    observations$survey == 0
-outer_model = zones$peripheral_municipality[m] &
-    observations$izone %in% zones$zone &
-    observations$jzone %in% zones$zone &
-    observations$survey == 2
-stopif(any(hs15_model & outer_model))
-
-observations$model = NA
-observations$model[hs15_model] = 1
-observations$model[outer_model] = 2
-
-# Subset estimation data
-
-observations = subset(observations, mode %in% 1:5)
-observations = subset(observations, !is.na(model))
-
-observations = downclass(observations)
-check.na(observations)
-save(observations, file="observations.RData")
+for (i in rows.along(input)) {
+    
+    message(sprintf("%d/%d: %s", i, nrow(input), input$file[i]))
+    tours = load1(input$file[i])
+    
+    observations = data.frame(pid=tours$pid)
+    observations$mode = tours$mode
+    observations$ttype = NA
+    m = grepl("^(1 - 2)", tours$tour_type, perl=TRUE)
```

Comment:
This also looks like something that could be put into a function, not necessarily for reuse but it could improve the readability of the code

Reply:
https://github.com/HSLdevcom/helmet-data-preprocessing/pull/12/commits/1d4a24e86d6f3b8b158a9831bf9ed0e5271a74f6 I also found a bug when fixing this so thanks! ;)